cpp/src/io/parquet/chunked_state.hpp

```diff
@@ -40,15 +43,24 @@ struct pq_chunked_state {
   /// Overall file metadata.  Filled in during the process and written during write_chunked_end()
   cudf::io::parquet::FileMetaData md;
   /// current write position for rowgroups/chunks
-  size_t current_chunk_offset;
+  std::size_t current_chunk_offset;
   /// optional user metadata
-  table_metadata const* user_metadata = nullptr;
-  /// only used in the write_chunked() case. copied from the (optionally) user supplied
-  /// argument to write_parquet_chunked_begin()
   table_metadata_with_nullability user_metadata_with_nullability;
   /// special parameter only used by detail::write() to indicate that we are guaranteeing
   /// a single table write.  this enables some internal optimizations.
-  bool single_write_mode = false;
+  table_metadata const* user_metadata;
+  /// only used in the write_chunked() case. copied from the (optionally) user supplied
+  /// argument to write_parquet_chunked_begin()
+  bool single_write_mode;
+
```

Comment:
This adds a default constructor (needed for the chunked path and/or python iirc), but also removes the default init for user_metadata and single_write_mode (@nvdbaranec should confirm, but that looks somewhat sketchy to me).

Reply:
I can add back the `nullptr` default for metatdata.